js/wallet.js

```diff
@@ -40,10 +104,126 @@ class JS extends Wallet {
     mockery.registerMock('../../../js/lib/request', {
       request: request
     })
+    mockery.registerMock('../../../js/settings', {
+      getSetting: (key) => {
+        switch (key) {
+          case settings.PAYMENTS_MINIMUM_VISITS:
+          {
+            return self.minimumVisits
+          }
+          case settings.PAYMENTS_MINIMUM_VISIT_TIME:
+          {
+            return self.minimumVisitTime
+          }
+          case settings.PAYMENTS_CONTRIBUTION_AMOUNT:
+          {
+            return self.defaultContribution
+          }
+          case settings.PAYMENTS_ENABLED:
+          case settings.PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED:
+          {
+            return true
+          }
+        }
+      }
+    })
 
     this.ledger = require('../browser-laptop/app/browser/api/ledger')
+    this.ledgerStatuses = require('../browser-laptop/app/common/constants/ledgerStatuses')
+    this.setStateFile()
+
+    this.resetWalletProperties = function () {
+      self.wallet = 'default-wallet'
+      self.stateKey = 'default-state'
+      self.setStateFile()
+
+      const stateFile = JSON.parse(self.stateFile)
+      const body = Immutable.fromJS(responses['/v2/wallet/balance'][self.wallet])
+
+      self.setState(self.ledger.onWalletProperties(self.state, body))
+      self.setState(self.setInfo(self.state, 'paymentId', stateFile.properties.wallet.paymentId))
+      self.setState(self.ledger.onInitRead(self.state, stateFile))
+    }
+
+    this.callback = sinon.stub(this.ledger, 'callback').callsFake(function (err, result, delayTime) {
+      if (err) {
+        return
+      }
+      if (typeof delayTime === 'undefined') {
+        delayTime = Math.floor(Math.random() * (5 * 60 * 1000))
+      }
+      self.setState(self.ledger.onCallback(self.state, result, delayTime))
+    })
+
     this.roundtrip = sinon.stub(this.ledger, 'roundtrip').callsFake(function (params, options, callback) {
-      roundtrip(params, options, callback)
+      roundtrip(params, options, callback, self.wallet)
+    })
+
+    this.getBalance = sinon.stub(this.ledger, 'getBalance').callsFake(function (state) {
+      self.setState(self.ledger.getPaymentInfo(state))
+      return self.state
+    })
+
+    this.recoverWalletCallback = sinon.stub(this.ledger, 'recoverWalletCallback').callsFake(function (error, result) {
+      if (error != null) {
+        self.resetWalletProperties()
```

Comment:
Why do we reset props here?

Reply:
This does something a little different than it's namesake, basically ensuring that we don't lose an old wallet that hasn't been properly recovered.

I'll re-evaluate this one and also look in to renaming